tests/unit/test_x509.cpp

```diff
@@ -568,3 +568,11 @@ TEST_F(X509Test, testGetSerialNumber)
     EXPECT_EQ("-42", negativeSerialCert.getSerialNumberDecimal());
     EXPECT_EQ(std::vector<uint8_t>{0x2a}, negativeSerialCert.getSerialNumberBinary());
 }
+
+TEST_F(X509Test, testCACheck)
```

Comment:
I would separate positive and negative tests and give them more self-documented names. E.g testIntermediateCertificateIsCA(), testExpiredCertificateIsNotCA(), testClientCertificateIsNotCa() or something.